src/logic/nfts/marketplace.ts

```diff
@@ -270,6 +270,7 @@ export function marketplaceShouldFetch(filters: NFTFilters) {
     filters.itemId ||
     filters.isWearableSmart ||
     filters.isOnRent ||
+    filters.rentalStatus !== undefined ||
```

Comment:
Careful here, I believe that as we're using `params.getList` to get the param as an array, this would never be `undefined`. It will be `[]` if no `rentalStatus` is present or an array with values otherwise. 
I believe that what you need here is
```suggestion
    filters.rentalStatus.length ||
```
or more verbose
```suggestion
    filters.rentalStatus.length > 0 ||
```
WDYT?